### PR TITLE
drivers: counter: Add support for rpi_pico timer

### PR DESCRIPTION
**Pull request by soburi**
Adds support for rpi_pico timer
    
Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

---

This PR need to modify HAL changes https://github.com/zephyrproject-rtos/hal_rpi_pico/pull/5.

### CODE REVIEW

**Comment by soburi** [comment]
@yonsch 
> Is the patch to the SDK necessary, or can the relevant function just be implemented in Zephyr to fit our needs?

As a basic policy, I would like to use Pico-SDK.
However, Pico-SDK has an IRQ vector table that it own, which conflicts with Zephyr's implementation.
Due to this issue, the Pico-SDK cannot be used without modification.
So I have removed the IRQ vector table manipulation from the Pico-SDK and changed the interface slightly to fit with Zephyr.
From another point of view, it can be said that the negative impact is small because it cannot be used in the first place without modification.

File: drivers/counter/counter_rpi_pico_timer.c [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+			    (DEVICE_DT_GET(node_id)), 0);                                          \
+		irq_enable((DT_IRQ_BY_IDX(node_id, idx, irq)));                                    \
+	} while (false);
+
+#define COUNTER_RPI_PICO_TIMER(inst)                                                               \
+	static void counter_irq_config##inst(void)                                                 \
+	{                                                                                          \
+		DT_INST_FOREACH_PROP_ELEM(inst, interrupt_names, RPI_PICO_TIMER_IRQ_ENABLE);       \
+	}                                                                                          \
+	struct counter_rpi_pico_timer_ch_data ch_data##inst[DT_NUM_IRQS(DT_DRV_INST(inst))];       \
```
Comment (nordic-krch) [inline drivers/counter/counter_rpi_pico_timer.c:197]: missing `static`

**Review by nordic-krch — COMMENTED** [review]
looks ok. one minor finding.

**Comment by soburi** [comment]
@yonsch 
Could you take a look please?

**Review by yonsch — DISMISSED** [review]

**Review by yonsch — APPROVED** [review]

**Comment by soburi** [comment]
@nordic-krch 
Could you take a look please?

**Review by carlescufi — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**